tests/actions/ReportTest.js

```diff
@@ -0,0 +1,73 @@
+import moment from 'moment';
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '../../src/ONYXKEYS';
+import * as Pusher from '../../src/libs/Pusher/pusher';
+import PusherConnectionManager from '../../src/libs/PusherConnectionManager';
+import CONFIG from '../../src/CONFIG';
+import {subscribeToReportCommentEvents} from '../../src/libs/actions/Report';
+import waitForPromisesToResolve from '../utils/waitForPromisesToResolve';
+
+describe('actions/Report', () => {
+    it('should subscribe to the private-user-accountID channel and handle reportComment events', () => {
+        const REPORT_ID = 1;
+        const ACTION_ID = 1;
+        const REPORT_ACTION = {
+            actionName: 'ADDCOMMENT',
+            automatic: false,
+            created: 'Nov 6 2020 9:14am PST',
+            message: [{type: 'COMMENT', html: 'Testing a comment', text: 'Testing a comment'}],
+            person: [{type: 'TEXT', style: 'strong', text: 'Test User'}],
+            shouldShow: true,
+            timestamp: moment().unix(),
+            actorAccountID: 1,
+            sequenceNumber: ACTION_ID,
+            isAttachment: false,
+            loading: false,
+        };
+
+        // When using the Pusher mock the act of calling Pusher.isSubscribed will create a
+        // channel already in a subscribed state. These methods are normally used to prevent
+        // duplicated subscriptions, but we don't need them for this test.
+        Pusher.isSubscribed = jest.fn().mockReturnValue(false);
+        Pusher.isAlreadySubscribing = jest.fn().mockReturnValue(false);
+
+        // Connect to Pusher
+        PusherConnectionManager.init();
+        Pusher.init({
+            appKey: CONFIG.PUSHER.APP_KEY,
+            cluster: CONFIG.PUSHER.CLUSTER,
+            authEndpoint: `${CONFIG.EXPENSIFY.URL_API_ROOT}api?command=Push_Authenticate`,
+        });
+
+        let reportActions;
+        Onyx.connect({
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${REPORT_ID}`,
+            callback: val => reportActions = val,
+        });
+
+        // Set up Onyx with some test user data
+        Onyx.set(ONYXKEYS.SESSION, {accountID: 1, email: 'test@test.com'});
+        return waitForPromisesToResolve()
+            .then(() => {
+                subscribeToReportCommentEvents();
+                return waitForPromisesToResolve();
+            })
```

Comment:
Same here. Is this the same?

```
.then(subscribeToReportCommentEvents)
```

Reply:
`subscribeToReportCommentEvents()` is a good example of something that doesn't return a `Promise` but does async work in the call to `Pusher.subscribe()`. So, it's not really the same since we want to wait until we are subscribed.

Reply:
Ah, I didn't realize it didn't return a promise. Let's leave it the way it is.